shopinvader_payment/components/payment_transaction_event_listerner.py

```diff
@@ -30,6 +28,8 @@ def _confirm_and_invalidate_session(self, sale_order):
         setattr(work, "shopinvader_backend", shopinvader_backend)
         res = work.component(usage="cart")._to_json(sale_order)
         response.set_store_cache("last_sale", res.get("data", {}))
+        response.set_store_cache("cart", {})
+        response.set_session("cart_id", 0)
```

Comment:
@Cedric-Pigeon @rousseldenis I don't understand why this is necessary. Where do you see `work.component(usage="cart")._to_json(sale_order)` calling `set_store_cache` ?

Reply:
Ok no I see it's done [here](https://github.com/shopinvader/odoo-shopinvader/blob/7e5425e081286f19704d5ce82646b224f5364a40/shopinvader/services/cart.py#L461-L462) in v10. I was looking at v13 which is different. However it's a bit surprising to me that _to_json would have side effects on the cache... :shrug: